src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -398,16 +420,92 @@ Status GcsActorManager::RegisterActor(
   }
 
   // Mark the callback as pending and invoke it after the actor has been successfully
-  // created.
+  // flushed to the storage.
   actor_to_register_callbacks_[actor_id].emplace_back(std::move(callback));
   RAY_CHECK(registered_actors_.emplace(actor->GetActorID(), actor).second);
 
+  const auto &owner_address = actor->GetOwnerAddress();
+  auto node_id = ClientID::FromBinary(owner_address.raylet_id());
+  auto worker_id = WorkerID::FromBinary(owner_address.worker_id());
+  RAY_CHECK(unresolved_actors_[node_id][worker_id].emplace(actor->GetActorID()).second);
+
   if (!actor->IsDetached() && worker_client_factory_) {
     // This actor is owned. Send a long polling request to the actor's
     // owner to determine when the actor should be removed.
     PollOwnerForActorOutOfScope(actor);
   }
 
+  // The backend storage is reliable in the future, so the status must be ok.
```

Comment:
```suggestion
  // The backend storage is supposed to be reliable, so the status must be ok.
```
The problem about `in the future` is that when we really make storage reliable someday, we will easily forget to change such comments.